bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/overlay/AccessibleToolItemBuilder.java

```diff
@@ -49,26 +53,26 @@ public AccessibleToolItemBuilder withToolTipText(String newToolTipText) {
 		return this;
 	}
 
-	public AccessibleToolItemBuilder withSelectionListener(SelectionListener newSelectionListener) {
-		this.selectionListener = newSelectionListener;
+	public AccessibleToolItemBuilder withShortcuts(List<KeyStroke> newShortcuts) {
+		this.shortcuts = newShortcuts;
 		return this;
 	}
 
-	public ToolItem build() {
-		ToolItem toolItem = accessibleToolBar.createToolItem(styleBits);
-
-		if (image != null) {
-			toolItem.setImage(image);
-		}
+	public AccessibleToolItemBuilder withOperation(Runnable newOperation) {
+		this.operation = newOperation;
+		return this;
+	}
 
-		if (toolTipText != null) {
-			toolItem.setToolTipText(toolTipText);
-		}
+	public ToolItem build() {
+		Objects.requireNonNull(image, "image for tool item must not be null"); //$NON-NLS-1$
+		Objects.requireNonNull(operation, "operation for tool item must not be null"); //$NON-NLS-1$
+		Objects.requireNonNull(toolTipText, "tooltip for tool item must not be null"); //$NON-NLS-1$
```

Comment:
Is there any particular reason to add these constraints?
An `AccessibleToolItemBuilder` does not inherently have an operation or a toolTipText attached and some ToolItems can, for example, not carry an image but rather have a text inside of them (https://archive.eclipse.org/eclipse/downloads/documentation/2.0/html/plugins/org.eclipse.platform.doc.isv/reference/api/org/eclipse/swt/widgets/ToolItem.html#setText(java.lang.String))

If we want to enforce these constraints, I am in favor of renaming our ToolItems to "AccessibleActionToolItem" or something better

Reply:
The only reason was to avoid that, by accident, one of the item initializations misses to set one of these properties. Since we don't have tests for tool items being properly initialized, this still gives you instant feedback about non properly initialized items.

But I agree that if we want to enforce this, the builder should simply require calling all the `with` methods to a static correctness guarantee instead of performing runtime checks. I have removed them in 38fa391.